### PR TITLE
[th/wireguard-routes]

### PR DESCRIPTION
**Pull request by thom311**
Branch to automatically add direct routes to the AllowedIPs ranges of the WireGuard peers.

And, I couldn't refrain from refactoring NMDevice a bit :)

### CODE REVIEW

File: src/devices/nm-device-private.h [hunk 1]
```diff
@@ -75,11 +75,12 @@ void nm_device_set_firmware_missing (NMDevice *self, gboolean missing);
 void nm_device_activate_schedule_stage1_device_prepare (NMDevice *device);
 void nm_device_activate_schedule_stage2_device_config (NMDevice *device);
 
-void nm_device_activate_schedule_ip4_config_result(NMDevice *device, NMIP4Config *config);
-void nm_device_activate_schedule_ip4_config_timeout (NMDevice *device);
+void nm_device_activate_schedule_ip_config_result (NMDevice *device,
+                                                   int addr_family,
+                                                   gpointer config /* (NMIPConfig *) */);
```
Comment (bengal) [inline src/devices/nm-device-private.h]: Why not a NMIPConfig * argument instead of the void pointer, and explicit casts where needed?

Comment (thom311) [reply src/devices/nm-device-private.h]: because most callers will actually be required to cast. Anyway, yes, maybe that's better. Fixing...

**Comment by lkundrak** [comment]
LGTM

**Comment by thom311** [comment]
Thanks. merged to master.

**Pull request closed without merging**